include/f_core/net/c_transciever.h

```diff
@@ -0,0 +1,25 @@
+#ifndef C_TRANSCIEVER_H
+#define C_TRANSCIEVER_H
+
+#include <cstddef>
+
+class CTransceiver {
+public:
+    CTransceiver() = default;
+
+    virtual int TransmitSynchronous(const void *data, size_t len) = 0;
+
+    virtual int ReceiveSynchronous(void *data, size_t len) = 0;
+
+    virtual int TransmitAsynchronous(const void *data, size_t len) = 0;
+
+    virtual int ReceiveAsynchronous(void *data, size_t len) = 0;
+
+    virtual int SetRxTimeout(int timeout) = 0;
+protected:
+    ~CTransceiver() = default;
```

Comment:
methinks that this ought to be a virtual destructor ([sauce](https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors)). TLDR, there are situations where not having it be virtual causes the child (actual implementation class) destructor to not be called